web_tree_date_search_extension/static/src/js/search_extension.js

```diff
@@ -0,0 +1,187 @@
+                            var elem = this.parentElement.parentElement.parentElement.className;
+                            var res = elem.split("oe_gantt_filter_from_");
+                            self.current_date_from[res[1]] = this.value === '' ? null : this.value;
+                            if (self.current_date_from[res[1]]){
+                                self.current_date_from[res[1]] = instance.web.parse_value(
+                                    self.current_date_from[res[1]], {
+                                    "widget": "date"
+                                });
+                            }
+                            self.do_search(self.last_domain, self.last_context, self.last_group_by);
+                        });
+                        this.$el.parent().find(
+                            '.oe_gantt_filter_to_' + date_field + ':last .oe_datepicker_master').change(function() {
+                            var elem = this.parentElement.parentElement.parentElement.className;
+                            var res = elem.split("oe_gantt_filter_to_");
+                            self.current_date_to[res[1]] = this.value === '' ? null : this.value;
+                            if (self.current_date_to[res[1]]){
+                                self.current_date_to[res[1]] = instance.web.parse_value(
+                                    self.current_date_to[res[1]], {
+                                    "widget": "date"
+                                });
+                            }
+                            self.do_search(self.last_domain, self.last_context, self.last_group_by);
+                        });
+                        this.on('edit:after', this, function () {
+                            self.$el.parent().find('.oe_gantt_filter_from_' + date_field + ':last').attr('disabled', 'disabled');
+                            self.$el.parent().find('.oe_gantt_filter_to_' + date_field + ':last').attr('disabled', 'disabled');
+                        });
+                        this.on('save:after cancel:after', this, function () {
+                            self.$el.parent().find('.oe_gantt_filter_from_' + date_field + ':last').removeAttr('disabled');
+                            self.$el.parent().find('.oe_gantt_filter_to_' + date_field + ':last').removeAttr('disabled');
+                        });
+                    }
+                }
+                else{
+                    $('.ui-toolbar:last').hide();
+                }
+                this.search_extension_loaded = true;
+            }
+            return tmp;
+        },
+        do_search: function(domain, context, group_by) {
+            var self = this;
+            this.last_domain = domain;
+            this.last_context = context;
+            this.last_group_by = group_by;
+            this.old_search = _.bind(this._super, this);
+            return self.search_by_selection();
```

Comment:
Why are you using `self` in this method? I think there is no closure issue, isn't it?

Also, as `search_by_selection` looks to be synchronize, why are you binding `_super`?